tests/test_integrations.py

```diff
@@ -46,6 +54,72 @@
     }
 }
 
+TEST_SFN_CONTEXT = {
+	'task_token': MOCK_TASK_TOKEN,
+	'sfn_context': {
+		'execution_id': MOCK_EXECUTION_ID,
+		'artifacts': {
+			'event': {
+				'id': MOCK_EVENT_ID,
+				'created_at': 'some_point_in_time_and_space',
+				'data_types': {},
+				'details': {
+                    "some": "randon text"
+				},
+				'event_type': 'Test Human Interaction Workflow',
+				'event_meta': {},
+				'investigation_id': MOCK_INVESTIGATION_ID,
+				'status_': 'open',
+				'is_duplicate': False
+			},
+			'execution_id': MOCK_EXECUTION_ID
+		},
+		'State_Config': {
+			'Name': MOCK_STATE_NAME,
+			'Parameters': {}
+		}
+	}
+}
+
+MOCK_DB_CONTEXT = {
+  "datetime": "some_point_in_time_and_space",
+  "execution_id": "mock_execution_id",
+  "investigation_id": "mock_investigation_id",
+  "results": {
+    'artifacts': {
+        'event': {
+            'id': MOCK_EVENT_ID,
+            'created_at': 'some_point_in_time_and_space',
+            'data_types': {},
+            'details': {
+                "some": "randon text"
+            },
+            'event_type': 'Test Human Interaction Workflow',
+            'event_meta': {},
+            'investigation_id': MOCK_INVESTIGATION_ID,
+            'status_': 'open',
+            'is_duplicate': False
+        },
+        'execution_id': MOCK_EXECUTION_ID
+    },
+    "errors": {},
+    "results": {}
+  }
+}
+
+def test_state_handler_with_task_token():
+    client = boto3.client('dynamodb')
+    #  Setup DB context for the state handler
+    client.put_item(
+        TableName=os.environ['SOCLESS_RESULTS_TABLE'],
+        Item=dict_to_item(MOCK_DB_CONTEXT,convert_root=False)
+    )
+
+    state_handler = StateHandler(TEST_SFN_CONTEXT, MockLambdaContext(), mock_integration_handler)
+    assert state_handler.execution_id == TEST_SFN_CONTEXT['sfn_context']['artifacts']['execution_id']
```

Comment:
The purpose of our test case here is to ensure that our `state_handler.context` property is outputting the correct values for the given input. So these assert statements should test that the `state_handler.context` property has the information we expect it to have.

Reply:
Definitely an Improvement, but where are the checks that ensure that `task_token` and `state_name` are actually in the `state_handler.context` dictionary?

The `state_handler.context` dictionary is the final output of the `StateHandler` class. Its what we want to confirm meets our expectations when the test case runs. 

The ideal test here is to compare that the generated `state.context` object matches what we expect it to be